sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3599,22 +3599,36 @@ object GpuOverrides extends Logging {
     "rapids.gpu.postColToRowProcessing")
 }
 
+// work around any GpuOverride failures
+object GpuOverrideUtil extends Logging {
+  def tryOverride(fn: SparkPlan => SparkPlan): SparkPlan => SparkPlan = { plan =>
+    val planOriginal = plan.clone()
+    try {
+      fn(plan)
+    } catch {
+      case t =>
+        logWarning("Failed to apply GPU overrides, falling back on the original plan: " + t, t)
```

Comment:
> probably don't need both +t and print exception (, t), 

I think it's helpful to add the exception message to the main log statement. Otherwise it may appear as 2 unrelated lines in systems like Splunk and SumoLogic 

> Also question here is do we want this Warning or Error and just quit... if someone is trying to use the plugin and it doesn't load they may want to know vs it going along and running on cpu and them having to find some warning in logs. 

that's a good question. We could differentiate between quit on fatal exception, and fall back for NonFatal ones

Reply:
I still also wonder if it shouldn't at least be option to fail on any exception.  I can see ignoring it in some ways but in our testing environment for instance we should just fail and in many other cases when user is expecting it to use plugin I would rather see it fail.  @jlowe  thoughts on this?

Reply:
In theory the tests should already catch the plugin not replacing parts of the plan, but I agree there should be a way to configure this to always fail rather than fallback.

Reply:
added a !TEST_CONF guard to fallback.

Reply:
I was thinking just a separate conf users could also set, but I guess this is fine for now.